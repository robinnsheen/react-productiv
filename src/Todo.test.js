import Todo from "./Todo";
import React from "react";
import { render } from "@testing-library/react";

const testTodo = {
  title: "testTitle",
  priority: 2,
  description: "test description"
};

describe("productiv app", function () {
  it("renders without crashing", function () {
    render(<Todo todo={testTodo}/>);
  });

  it("renders with correct info", function () {
    const {container} = render(<Todo todo={testTodo} />);
    expect(container.querySelector(".Todo")).toContainHTML(
      `<div class="Todo">`+
        `<div>`+
          `<b>testTitle</b>`+
         ` <small>(priority: 2)</small>`+
        `</div><div>`+
          `<small>test description</small>`+
        `</div>`+
      `</div>`

    );
  });
  it("matches snapshot", function () {
    const { container } = render(<Todo todo={testTodo} />);
    expect(container).toMatchSnapshot()
  })
});
