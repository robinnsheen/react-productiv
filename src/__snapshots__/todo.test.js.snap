// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv app matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        testTitle
      </b>
       
      <small>
        (priority: 
        2
        )
      </small>
    </div>
    <div>
      <small>
        test description
      </small>
    </div>
  </div>
</div>
`;
